catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/util/EndpointUtil.java

```diff
@@ -586,18 +587,36 @@ public Serializable parseDate(Serializable value) {
 
     String string = String.valueOf(value);
 
+    if (StringUtils.isBlank(string)) {
+      return null;
+    }
+
+    if (StringUtils.isNumeric(string)) {
+      try {
+        return Instant.ofEpochMilli(Long.parseLong(string));
+      } catch (NumberFormatException ex) {
+        throw new IllegalArgumentException(ex);
+      }
+    }
+
+    if (iso8601Z.matcher(string).matches()) {
+      return Instant.parse(string);
+    }
+
+    if (iso8601Offset.matcher(string).matches()) {
+      return OffsetDateTime.parse(string).toInstant();
+    }
+
     SimpleDateFormat dateFormat;
     if (boonDefault.matcher(string).matches()) {
       dateFormat = new SimpleDateFormat("EEE MMM d HH:mm:ss zzz yyyy");
-    } else if (iso8601.matcher(string).matches()) {
-      dateFormat = new SimpleDateFormat(ISO_8601_DATE_FORMAT);
     } else {
       dateFormat = new SimpleDateFormat();
     }
     dateFormat.setTimeZone(TimeZone.getTimeZone("UTC"));
 
     try {
-      return dateFormat.parse(value.toString());
+      return dateFormat.parse(value.toString()).toInstant();
```

Comment:
❗️ This was an error from the original code, but we should be using the `string` variable from line 588 that was already created instead of doing `value.toString()`